lib/promiscuous/subscriber/unit_of_work.rb

```diff
@@ -36,12 +36,25 @@ def self.current=(value)
   end
 
   def process_message
+    retries = 0
+    retry_max = 50
+
     begin
       on_message
     rescue Exception => e
       Promiscuous::Config.error_notifier.call(e)
-      message.nack
       raise e if Promiscuous::Config.test_mode
+
+      # TODO: use dependency injection here instead
```

Comment:
i like the approach we were talking about of moving `process_message` into each backend